metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishedRecordDao.java

```diff
@@ -147,59 +150,107 @@ long countDepublishedRecordsForDataset(String datasetId) {
    */
-  public void markAllAsNotDepublished(String datasetId){
+  public Set<String> getAllDepublishedRecords(String datasetId,
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
+    final Query<DepublishedRecord> query = prepareQueryForDepublishedRecords(datasetId, sortField,
+        sortDirection, depublicationState, searchQuery);
+    query.project(DepublishedRecord.RECORD_ID_FIELD, true);
+    query.project(DepublishedRecord.ID_FIELD, false);
+
+    // Execute query with correct pagination
+    final List<DepublishedRecord> result = getListOfQuery(query, null);
+
+    // Convert result to right object.
+    return result.stream().map(DepublishedRecord::getRecordId).collect(Collectors.toSet());
+  }
 
+  private Query<DepublishedRecord> prepareQueryForDepublishedRecords(String datasetId,
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
     // Create query.
     final Query<DepublishedRecord> query = morphiaDatastoreProvider.getDatastore()
-            .createQuery(DepublishedRecord.class);
+        .createQuery(DepublishedRecord.class);
     query.field(DepublishedRecord.DATASET_ID_FIELD).equal(datasetId);
+    if (Objects.nonNull(depublicationState)) {
+      query.field(DepublishedRecord.DEPUBLICATION_STATE_FIELD).equal(depublicationState);
+    }
+    if (StringUtils.isNotBlank(searchQuery)) {
+      query.field(DepublishedRecord.RECORD_ID_FIELD).contains(searchQuery);
+    }
+
+    // Set ordering
+    query.order(sortDirection.createSort(sortField.getDatabaseField()));
+    return query;
+  }
+
+  /**
+   * This method marks record ids with the provided {@link DepublicationStatus} and {@link Date}
+   * where appropriate.
+   * <p>A {@link DepublicationStatus#PENDING_DEPUBLICATION} unsets the depublication date</p>
+   * <p>A {@link DepublicationStatus#DEPUBLISHED} sets the depublication date with the one
+   * provided</p>
+   *
+   * @param datasetId the dataset for which to do this
+   * @param depublicationStatus the depublication status. Null status is translated to {@link
+   * DepublicationStatus#PENDING_DEPUBLICATION}
+   * @param depublicationDate the depublication date. Can be null
+   */
+  public void markRecordIdsWithDepublicationStatus(String datasetId,
+      DepublicationStatus depublicationStatus, @Nullable Date depublicationDate) {
+
+    // Create query.
+    final Query<DepublishedRecord> query = morphiaDatastoreProvider.getDatastore()
+        .createQuery(DepublishedRecord.class);
+    query.field(DepublishedRecord.DATASET_ID_FIELD).equal(datasetId);
+    //Convert status if provided status is null
+    final DepublicationStatus depublicationStatusChecked =
+        depublicationStatus == null ? DepublicationStatus.PENDING_DEPUBLICATION
+            : depublicationStatus;
```

Comment:
This seems overkill? Why not require that this parameter is non-null? It is never  called with null anyway.

Reply:
Changing but give it another thought.. Do we accept null values to go in the db?
Same goes for the depublicationDate which is not safe guarded either.

Reply:
Then in my view we should have a sanity check for null in the method. This way we force the calling method to do it's homework too (and not just pass null, which is not clear to the reader). 
But I agree with you that it's subjective and that there are arguments for your approach as well.
I leave it up to you.